File: leetcode/009-palindrome-number/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        long reversed = 0;
        int original = x;
        while (x != 0) {
            reversed = reversed * 10 + x % 10;
            x /= 10;
        }
        return original == reversed;
    }
};

int main() {
    Solution sol;
    cout << sol.isPalindrome(121) << endl;  // 1 (true)
    cout << sol.isPalindrome(-121) << endl; // 0 (false)
    cout << sol.isPalindrome(10) << endl;   // 0 (false)
    return 0;
}

File: leetcode/009-palindrome-number/solution.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        reversed_num = 0
        original = x
        while x != 0:
            reversed_num = reversed_num * 10 + x % 10
            x //= 10
        return original == reversed_num


if __name__ == "__main__":
    print(Solution().isPalindrome(121))   # True
    print(Solution().isPalindrome(-121))  # False
    print(Solution().isPalindrome(10))    # False
